src/test/java/org/jabref/logic/cleanup/RemoveLinksToNotExistentFilesTest.java

```diff
@@ -0,0 +1,165 @@
+    private Path defaultFileFolder;
+    private Path fileBefore;
+    private BibEntry entry;
+    private FilePreferences filePreferences;
+    private BibDatabaseContext databaseContext;
+    private RemoveLinksToNotExistentFiles removeLinks;
+
+    @BeforeEach
+    void setUp(@TempDir Path bibFolder) throws IOException {
+        // The folder where the files should be moved to
+        defaultFileFolder = bibFolder.resolve("pdf");
+        Files.createDirectory(defaultFileFolder);
+
+        // The folder where the files are located originally
+        Path fileFolder = bibFolder.resolve("files");
+        Files.createDirectory(fileFolder);
+        fileBefore = fileFolder.resolve("test.pdf");
+        Files.createFile(fileBefore);
+
+        MetaData metaData = new MetaData();
+        metaData.setDefaultFileDirectory(defaultFileFolder.toAbsolutePath().toString());
+        databaseContext = new BibDatabaseContext(new BibDatabase(), metaData);
+        Files.createFile(bibFolder.resolve("test.bib"));
+        databaseContext.setDatabasePath(bibFolder.resolve("test.bib"));
+
+        LinkedFile fileField = new LinkedFile("", fileBefore.toAbsolutePath(), "");
+
+        // Entry with one online and one normal linked file
+        entry = new BibEntry(StandardEntryType.Article)
+                .withField(StandardField.AUTHOR, "Shatakshi Sharma and Bhim Singh and Sukumar Mishra")
+                .withField(StandardField.DATE, "April 2020")
+                .withField(StandardField.YEAR, "2020")
+                .withField(StandardField.DOI, "10.1109/TII.2019.2935531")
+                .withField(StandardField.FILE, FileFieldWriter.getStringRepresentation(Arrays.asList(
+                    new LinkedFile("", "https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8801912:PDF", ""),
+                    fileField)))
+                .withField(StandardField.ISSUE, "4")
+                .withField(StandardField.ISSN, "1941-0050")
+                .withField(StandardField.JOURNALTITLE, "IEEE Transactions on Industrial Informatics")
+                .withField(StandardField.PAGES, "2346--2356")
+                .withField(StandardField.PUBLISHER, "IEEE")
+                .withField(StandardField.TITLE, "Economic Operation and Quality Control in PV-BES-DG-Based Autonomous System")
+                .withField(StandardField.VOLUME, "16")
+                .withField(StandardField.KEYWORDS, "Batteries, Generators, Economics, Power quality, State of charge, Harmonic analysis, Control systems, Battery, diesel generator (DG), distributed generation, power quality, photovoltaic (PV), voltage source converter (VSC)");
+
+        filePreferences = mock(FilePreferences.class);
+        when(filePreferences.shouldStoreFilesRelativeToBibFile()).thenReturn(false);
+        removeLinks = new RemoveLinksToNotExistentFiles(databaseContext, filePreferences);
+    }
+
+    @Test
+    void deleteFileInMultipleLinkedEntry() {
```

Comment:
Maybe a slightly better method name:

```suggestion
    void deleteFileInEntryWithMultipleFileLinks() {
```